server/src/main/java/org/elasticsearch/plugins/SystemIndexPlugin.java

```diff
@@ -49,4 +60,40 @@
     default Collection<String> getAssociatedIndexPatterns() {
         return Collections.emptyList();
     }
+
+    /**
+     * Cleans up the state of the feature by deleting system indices and associated indices.
+     * Override to do more for cleanup (e.g. cancelling tasks).
+     * @param clusterService Cluster service to provide cluster state
+     * @param client A client, for executing actions
+     * @param listener Listener for post-cleanup result TODO[wrb]: need to gather results over features
+     */
+    default void cleanUpFeature(
```

Comment:
I added a check for whether or not there are indices to delete before issuing the request, but also moved the logic to `SystemIndices.Feature#cleanUpFeature` to make it available to built-in features rather than just plugins.